### PR TITLE
Allow kuzzle to handle HTTP forms again

### PR DESCRIPTION
**Pull request by scottinet**
# Description

HTTP forms (both `application/x-www-form-urlencoded` and `multipart/form-data`) weren't accepted anymore by Kuzzle, whereas all the necessary code was there. This was because of an erroneous message validation check, which was only considering `application/json` queries.

# Other changes

There were no tests to verify that HTTP forms are accepted. Functional tests have been added to prevent regressions from occuring again.

### CODE REVIEW

File: lib/core/network/protocols/http.js [hunk 1]
```diff
@@ -78,6 +85,30 @@ class HttpMessage {
     }
   }
 
+  /**
+   * Checks that an incoming HTTP message is well-formed
+   *
+   */
+  validate () {
```
Comment (Aschen) [inline lib/core/network/protocols/http.js:92]: Where this method is called? I can't find the call anywhere :thinking:

Comment (scottinet) [reply lib/core/network/protocols/http.js:92]: ask my brain :expressionless:

File: lib/core/network/protocols/http.js [hunk 2]
```diff
@@ -78,7 +85,33 @@ class HttpMessage {
...
+    if (this.isEmpty()) {
+      return;
+    }
+
+    const contentType = this.headers['content-type'];
```
Comment (Aschen) [inline lib/core/network/protocols/http.js:97]: Should we consider the absence of `content-type` as an error?

Comment (scottinet) [reply lib/core/network/protocols/http.js:97]: no, we default it to `application/json`

**Review by Aschen — APPROVED** [review]

**Review by rolljee — APPROVED** [review]

**Review by xbill82 — APPROVED** [review]

**Pull request merged**